File: Day-02/cpp/GameduJS/Day02.cpp
#include <cstring>
#include "iostream"
#include "fstream"
#include "vector"

int x = 0;
int y = 0;
int aim = 0;

std::vector<std::string> split(std::string string, std::string delimiter);

int main()
{
    std::vector<std::string> commandCollection;

    std::ifstream stream("../shared/Day02/CommandInput.txt");

    if(!stream.is_open())
        return 0;

    std::string line;
    while(std::getline(stream, line))
    {
        commandCollection.push_back(line);
    }

    for(int i = 0; i < commandCollection.size(); ++i)
    {
        std::string commands = commandCollection[i];
        std::vector<std::string> commandSplit = split(commands, " ");

        std::string command = commandSplit[0];
        int commandValue = std::stoi(commandSplit[1]);

        if(command == "forward")
        {
            x+=commandValue;
        }
        if(command == "down")
        {
            y+=commandValue;
        }
        if(command == "up")
        {
            y-=commandValue;
        }

    }

    std::cout << " Part 1 solution: " << x * y << std::endl;
    x = 0;
    y = 0;

    // Part 2
    for(int i = 0; i < commandCollection.size(); ++i)
    {
        std::string commands = commandCollection[i];
        std::vector<std::string> commandSplit = split(commands, " ");

        std::string command = commandSplit[0];
        int commandValue = std::stoi(commandSplit[1]);

        if(command == "down")
            aim+=commandValue;
        if(command == "up")
            aim-=commandValue;
        if(command == "forward")
        {
            x+=commandValue;
            y+=aim * commandValue;
        }
    }

    std::cout << "Part 2 solution: " << x * y << std::endl;




    return 0;
}

std::vector<std::string> split(std::string string, std::string delimiter)
{
    std::vector<std::string> vector;

    size_t pos = 0;
    std::string token;
    if ((pos = string.find(delimiter)) != std::string::npos) {
        token = string.substr(0, pos);
        vector.push_back(token);
        vector.push_back(string.substr(token.size() + 1));
    }

    return vector;
}

File: Day-02/python/paula003/day02.py
def task1():
    depth = 0
    horizontal = 0
    with open('./day02.txt') as file:
        for line in file:
            cmd = line.split(' ')[0]
            value = int(line.split(' ')[1])
            if cmd == 'forward':
                horizontal += value
            elif cmd == 'up':
                depth -= value
            elif cmd == 'down':
                depth += value
    print('Lösung der 1. Aufgabe: {}'.format(depth * horizontal))


def task2():
    depth = 0
    horizontal = 0
    aim = 0
    with open('./day02.txt') as file:
        for line in file:
            cmd = line.split(' ')[0]
            value = int(line.split(' ')[1])
            if cmd == 'forward':
                horizontal += value
                depth += aim * value
            elif cmd == 'up':
                aim -= value
            elif cmd == 'down':
                aim += value
    print('Lösung der 2. Aufgabe: {}'.format(depth * horizontal))


task1()
task2()
